File: ceylon/src/main/java/com/laamella/code_state_machine/Actions.java
package com.laamella.code_state_machine;

import com.laamella.code_state_machine.util.Chain;

/**
 * A simple wrapper around a list of actions.
 */
public class Actions extends Chain<Action> {
	public Actions(final Action... actions) {
		super(actions);
	}

	public void execute() {
		for (final Action action : getItems()) {
			action.execute();
		}
	}
}

File: kotlin/src/main/java/com/laamella/kode_state_machine/Actions.kt
package com.laamella.kode_state_machine

import com.laamella.kode_state_machine.util.Chain

/**
 * A simple wrapper around a list of actions.
 */
class Actions(vararg actions: Action) : Chain<Action>(*actions) {
    fun execute() {
        for (action in getItems()) {
            action.execute()
        }
    }
}
